### ISSUE TITLE
[FR]: Misused of DisposableEffect instead of LaunchedEffect

### ISSUE THREAD

**Issue by mhdabbaghy** [android/nowinandroid#515]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the problem

in MainActivity in order to update the dark content of the system bars to match the theme, there is a DisposableEffect which `onDispose` method is empty with no comment

`
DisposableEffect(systemUiController, darkTheme) {
    systemUiController.systemBarsDarkContentEnabled = !darkTheme
    onDispose {}
}
`

### Describe the solution

I can update it and use LaunchedEffect instead

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hoc081098** [comment]
I think DisposableEffect is more suitable. LaunchedEffect is used to call suspend function

**Comment by mhdabbaghy** [comment]
I do agree with you about not using suspend function but we are not disposing anything at all

**Comment by SimonMarquis** [comment]
I also think this is the right usage of `DisposableEffect` based on it's javadoc.
The `onDispose` method is empty because there is not special code associated to this callback, but it has to be returned from the `DisposableEffect` call.
This issue should be closed, unless you have additionnal feedback :)

**Comment by dev-weiqi** [comment]
@SimonMarquis It's a bit of a conflict with the official docs 👀

<img width="858" alt="截圖 2023-10-08 00 05 34" src="https://github.com/android/nowinandroid/assets/55677874/940037c4-9b51-4a8b-8d78-bdfccdbbc101">

**Comment by alexvanyo** [comment]
Agreed that this is not an ideal use of `DisposableEffect`, as it leads to an empty `onDispose` which is generally to be avoided.

The reason `DisposableEffect` is used over `LaunchedEffect` in this particular case is to update the system UI for a new dark mode setting as soon as possible: If the code was in a `LaunchedEffect`, it would run slightly later due to dispatching than when it runs now when it is placed in a `DisposableEffect`, and window flags are especially sensitive to timing.
